code/game/machinery/dance_machine.dm

```diff
@@ -386,16 +412,20 @@
 		sleep(0.1 SECONDS)
 	M.lying_fix()
 
-/obj/machinery/jukebox/disco/proc/dance4(mob/living/M)
+/obj/machinery/jukebox/disco/proc/dance4(mob/living/lead_dancer)
 	var/speed = rand(1,3)
 	set waitfor = 0
 	var/time = 30
 	while(time)
 		sleep(speed)
 		for(var/i in 1 to speed)
-			M.setDir(pick(GLOB.cardinals))
-			for(var/mob/living/carbon/NS in rangers)
-				NS.set_resting(!NS.resting, TRUE, TRUE)
+			lead_dancer.setDir(pick(GLOB.cardinals))
+			// makes people dance with us nearby
+			for(var/datum/weakref/weak_dancer as anything in rangers)
+				var/mob/living/carbon/dancer = weak_dancer.resolve()
+				if(!istype(dancer))
+					continue
+				dancer.set_resting(!dancer.resting, silent = TRUE, instant = TRUE)
```

Comment:
Did anyone actually consider the consequence of this, lol?

Every mob has a 33% chance of rolling `dance4` which in turn makes all listening spam rest
With just six listeners it's *more* than likely people are spamming rest multiple times a process tick

Reply:
funny caramaldansen song dancing :)